add validation for linkedin url

arslanashraf7 opened the pull request:
#### What are the relevant tickets?
#1246 

#### What's this PR do?
Adds two types of validation on the LinkedIn URLs:
- The length of the URL is not more than 200
- The URL format is correct for a profile

#### How should this be manually tested?
- Run the app
- Add a Linkedin URL with more than 200 characters (You should see an error message)
- Add a non-LinkedIn or ill-formatted LinkedIn URL (You should see an error message)
- You should be able to verify the LinkedIn URL pattern introduced in this PR.

#### Where should the reviewer start?
- On Home page click `Apply Now`
- Next step, Click `View/Edit Resume or LinkedIn Profile`
- Next step, Add URLs in the linked in the field

#### Any background context you want to provide?
Some users were adding URLs having a length of more than 200 characters.

#### Screenshots (if appropriate)
**When URL is not valid (If its is not a linked in or URL or an invalid linkedin URL)**
<img width="559" alt="Screenshot 2021-07-07 at 5 49 35 PM" src="https://user-images.githubusercontent.com/34372316/124763280-3a636380-df4d-11eb-99b8-e2d4e60784ce.png">

**When URL length is more then 200**
<img width="600" alt="Screenshot 2021-07-07 at 5 50 29 PM" src="https://user-images.githubusercontent.com/34372316/124763301-3d5e5400-df4d-11eb-940a-6901229cc23a.png">

pdpinch commented on applications/views.py:
```diff
@@ -218,6 +221,22 @@ def post(self, request, *args, **kwargs):
+                {"errors": "The URL should be of 200 characters at max"}
+            )
+
+        regex = re.compile("^https://[a-z]{2,3}[.]linkedin[.]com/.*$", re.I)
```
I think it makes sense to check the URL for a linkedin domain name, but are you sure that they use a shortened or localized domain? 

We should at least take a look at the data we've already collected from users.

arslanashraf7 replied:
Correct, We can take a look at the collected data and base this logic on that. I can verify that data on RC would that be enough? or would it be a good idea to take a look at a more realistic data set?

pdpinch commented on applications/views.py:
```diff
@@ -218,6 +221,22 @@ def post(self, request, *args, **kwargs):
+        """
+        if len(linkedin_url) > 200:
+            raise ValidationError(
+                {"errors": "The URL should be of 200 characters at max"}
```
```suggestion
                {"errors": "The URL should be less than 200 characters."}
```

Trying to keep the message as simple as possible.

briangrossman commented:
@arslanashraf7 
After reviewing the data on prod, discussing the option of loosening the restrictions, and talking with Mariah, let's proceed with the path you were on of checking for a valid LinkedIn URL. In Slack you said we could use this logic:

`http<s>://<www or country_code or empty>.linkedin.<com or country_code>/in/<any_value(e.g. Username)>`

This seems pretty good, but check out rows 4627 through 4734 which contain /mwlite/ in the URL (and seem to work). Also, it looks like you can use a /m/ as well:

For example:
- https://www.linkedin.com/mwlite/in/brian-grossman-705289
- https://www.linkedin.com/m/in/brian-grossman-705289

Thanks!

arslanashraf7 commented:
@briangrossman Thanks for giving a thorough look at the data and identifying one other valid the usecase. 

>This seems pretty good, but check out rows 4627 through 4734 which contain /mwlite/ in the URL (and seem to work). Also, it looks like you can use a /m/ as well:

I've updated the PR with an updated LinkedIn URL validation pattern and I've also added some screenshots. Furthermore, I'm setting up the CI build for this PR so that you can take a look flexibly if you'd like.

briangrossman commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Brian Grossman](https://trello.com/c/fG9vk9pv/6241-brian-grossman)

briangrossman commented:
@arslanashraf7 
I did a bunch of testing on the CI build using real data. It worked great. Functionally, I'm giving it a 👍

HamzaIbnFarooq reviewed (approved):
👍
